InnerEye/ML/Histopathology/utils/heatmap_utils.py

```diff
@@ -0,0 +1,49 @@
+from typing import Optional, List
+import matplotlib.pyplot as plt
+import numpy as np
+
+from matplotlib.axes import Axes
+from matplotlib.image import AxesImage
+import matplotlib.patches as patches 
+import matplotlib.collections as collection
+
+
+def plot_heatmap_selected_tiles(tile_coords: np.array,
+                                tile_values: np.ndarray,
+                                location_bbox: List[int],
+                                tile_size: int,
+                                level: int,
+                                ax: Optional[Axes] = None) -> AxesImage:
+    """Plots a 2D heatmap for selected tiles to overlay on the slide.
+    :param tile_coords: XY tile coordinates, assumed to be spaced by multiples of `tile_size` (shape: [N, 2]).
+    :param tile_values: Scalar values of the tiles (shape: [N]).
+    :param location_bbox: Location of the bounding box of the slide.
+    :param level: Magnification at which tiles are available (e.g. PANDA levels are 0 for original, 1 for 4x downsampled, 2 for 16x downsampled).
+    :param tile_size: Size of each tile.
+    :param ax: Axes onto which to plot the heatmap (default: current axes).
+    """
+    if ax is None:
+        ax = plt.gca()
+
+    level_dict = {"0": 1, "1": 4, "2": 16}
+    factor = level_dict[str(level)]
```

Comment:
why do you not make the level_dict keys integers right away?

Reply:
For PANDA, these are the mappings of the level and the corresponding downscaling factors.

Reply:
```suggestion
    level_dict = {0: 1, 1: 4, 2: 16}
    factor = level_dict[level]
```

Reply:
Added a code suggestion to clarify

Reply:
Thanks, I changed it now